src/libstd/macros.rs

```diff
@@ -444,9 +444,16 @@ pub mod builtin {
     #[macro_export]
     macro_rules! cfg { ($($cfg:tt)*) => ({ /* compiler built-in */ }) }
 
-    /// Parse the current given file as an expression.
-    ///
-    /// This is generally a bad idea, because it's going to behave unhygienically.
+    /// Parse the file provided in the argument as an expression, item, or
+    /// statement according to the context. This file is located relative to the
+    /// current file (similarly to how modules are found).
+    ///
+    /// Using this macro is often a bad idea, because the file is
+    /// going to be parsed and placed in the surrounding code
+    /// unhygenically. This could result in variables or functions
+    /// being different from what the file expected if there are
+    /// variables or functions that have the same name in the current
+    /// file.
```

Comment:
If the file as parsed as an item, I believe hygiene is never an issue (that is, using `include!`ing items is equivalent to copying and pasting source code).